packages/jest-changed-files/src/sl.ts

```diff
@@ -42,8 +45,29 @@ const adapter: SCMAdapter = {
   },
 
   getRoot: async cwd => {
+    if (isSteamLocomotive) {
+      return null;
+    }
+
     try {
-      const result = await execa('sl', ['root'], {cwd, env});
+      const subprocess = execa('sl', ['root'], {cwd, env});
+
+      // Check if we're calling sl (steam locomotive) instead of sl (sapling)
+      // by looking for the escape character in the first chunk of data.
+      if (subprocess.stdout) {
+        subprocess.stdout.once('data', (data: Buffer | string) => {
+          data = Buffer.isBuffer(data) ? data.toString() : data;
+          if (data.codePointAt(0) === 27) {
+            subprocess.cancel();
+            isSteamLocomotive = true;
+          }
+        });
+      }
+
+      const result = await subprocess;
+      if (result.killed) {
```

Comment:
```suggestion
      if (result.killed && isSteamLocomotive) {
```

?

Just in case it's killed for some other reason?

Reply:
Makes sense to me!